dev-libs/lief/lief-0.11.5.ebuild

```diff
@@ -0,0 +1,125 @@
+#    invoking CMake in mostly non-configurable (and thus broken) ways.
+# We choose the working approach.
+inherit python-single-r1 cmake
+
+DESCRIPTION="Library to instrument executable formats"
+HOMEPAGE="https://lief.quarkslab.com"
+SRC_URI="https://github.com/lief-project/LIEF/archive/${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="Apache-2.0"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="c examples +python static-libs"
+REQUIRED_USE="python? ( ${PYTHON_REQUIRED_USE} )"
+
+# See "cmake/LIEFDependencies.cmake" for C and C++ dependencies.
+BDEPEND="
+	python? (
+		$(python_gen_cond_dep '
+			>=dev-python/setuptools-31.0.0[${PYTHON_USEDEP}]
+		')
+	)
+"
+#FIXME: Add after bumping to the next stable release:
+#	>=dev-libs/spdlog-1.8.5
+RDEPEND="python? ( ${PYTHON_DEPS} )"
+DEPEND="${RDEPEND}"
+
+# LIEF tests are non-trivial (if not infeasible) to run in the general case.
+# For example, "tests/CMakeLists.txt" implies all USE flags must be enabled:
+#     if (NOT LIEF_ELF OR NOT LIEF_PE OR NOT LIEF_MACHO)
+#       message(FATAL_ERROR "Tests require all LIEF's modules activated" )
+#     endif()
+RESTRICT="test"
+
+S="${WORKDIR}/LIEF-${PV}"
+
+DISTUTILS_OPTIONAL=1
+
+pkg_setup() {
+	use python && python-single-r1_pkg_setup
+}
+
+#FIXME: Unvender currently vendored dependencies in "third-party/". Ideally,
+#upstream should add one "LIEF_EXTERNAL_${LIBNAME}" CMake option governing each
+#vendored dependency resembling the existing "LIEF_EXTERNAL_SPDLOG" option.
+#Note that LIEF patches the vendored "Boost leaf" and "utfcpp" dependencies.
+src_prepare() {
+	# Respect "multilib"-based lib dirnames.
+	sed -i -e 's~\bDESTINATION lib\(64\)\{0,1\}\b~DESTINATION ${CMAKE_INSTALL_LIBDIR}~' \
+		CMakeLists.txt || die
+
+	# Respect "python"-based installation of Python bindings. Upstream
+	# currently fails to install these bindings, resulting in Gentoo "RUNPATH"
+	# QA notices at installation time. See also:
+	#    https://github.com/lief-project/LIEF/issues/599#issuecomment-889654343
+	cat <<- EOF >> api/python/CMakeLists.txt || die
+		if(LIEF_INSTALL_PYTHON)
+		  install(
+		    TARGETS pyLIEF
+		    DESTINATION "$(python_get_sitedir)"
+		    COMPONENT libraries
+		  )
+		endif()
+	EOF
+
+	cmake_src_prepare
+}
+
+src_configure() {
+	# See also:
+	# * "cmake/LIEFDependencies.cmake" for a dependency list.
+	# * "cmake/LIEFOptions.cmake" for option descriptions.
+	local mycmakeargs=(
+		-DLIEF_COVERAGE=OFF
+		-DLIEF_DISABLE_FROZEN=OFF
+		-DLIEF_EXTRA_WARNINGS=OFF
+		-DLIEF_PROFILING=OFF
+		-DLIEF_SUPPORT_CXX14=ON
+		-DLIEF_USE_CCACHE=OFF
+
+		# Disabling LIEF's format options causes build failures. See also:
+		#    https://github.com/lief-project/LIEF/issues/599
+		-DLIEF_ELF=ON
+		-DLIEF_PE=ON
+		-DLIEF_MACHO=ON
+		-DLIEF_ART=ON
+		-DLIEF_DEX=ON
+		-DLIEF_OAT=ON
+		-DLIEF_VDEX=ON
+
+		-DBUILD_SHARED_LIBS="$(usex static-libs OFF ON)"
+		-DLIEF_C_API="$(usex c ON OFF)"
+		-DLIEF_EXAMPLES="$(usex examples ON OFF)"
+		-DLIEF_FORCE32="$(usex x86 ON OFF)"
```

Comment:
Is this a x86 thing, or is it a multilib thing? Because I suspect the latter, in which case the `cmake-multilib` eclass might come in handy. I haven't ever used it myself, but the description says "provides a glue between cmake.eclass(5) and multilib-minimal.eclass(5)" so it does sound like it might do what we need it to do.

Reply:
**Oh.** I didn't even know `cmake-multilib` was a thing! Upstream's CMake support is non-standard and seemingly untested. So, I must confess I actually have no idea what (if anything) `-DLIEF_FORCE32=ON` is doing.

Let's see how far down the rabbit hole I get before throwing up.

Reply:
**Fascinating.** `-DLIEF_FORCE32=ON` just internally passes `-m32` to the linker. There's probably a *much* saner way for us to do that, so let's see what `cmake-multilib` has to say about this.